llama-index-packs/llama-index-packs-raft-dataset/llama_index/packs/raft_dataset/base.py

```diff
@@ -0,0 +1,210 @@
+"""RAFT Dataset LlamaPack class."""
+
+# Inspired from https://github.com/ShishirPatil/gorilla/tree/main/raft
+
+from typing import Any
+import random
+from datasets import Dataset
+
+from llama_index.core.llama_pack.base import BaseLlamaPack
+from llama_index.core import SimpleDirectoryReader
+
+# from llama_index.core.node_parser import SemanticSplitterNodeParser
+from llama_index.core.llms import ChatMessage
+from llama_index.llms.openai import OpenAI
+from llama_index.core.node_parser import SimpleNodeParser
+
+DEFAULT_CHUNK_SIZE = 512
+# DEFAULT_BREAKPOINT_PERCENTILE_THRESHOLD = 95 # Use when SemanticSplitterNodeParser
+
+
+class RAFTDatasetPack(BaseLlamaPack):
```

Comment:
it seems only getting dataset as from https://github.com/ShishirPatil/gorilla/pull/218/files
do we also have fine tuning ?

Reply:
Yeah. Once, we have the dataset we can use HF/ Unsloth/ any other framework for finetuning. They have used Azure platform for finetuning.